### PR TITLE
chore: Add smart wallet utility functions

### PR DESCRIPTION
**Pull request by cpcramer**
**What changed? Why?**
Add Smart Wallet helper utility. 

- This excludes the main components and functionality which will be added in a follow up PR. 
- Adds the `Read Me` section that will live below the demo. It's a copy of the Smart Wallet Paymaster documentation which can be found [here](https://www.smartwallet.dev/guides/paymasters). 
- Add `isLocal` utility function + unit tests.
- Fix broken /paymaster-bundler links. Add temp coming soon landing page

**Notes to reviewers**

**How has it been tested?**

### CODE REVIEW

File: template/web/app/new-paymaster-bundler/_components/CallStatus.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    data?: {
+      status?: string;
+    };
+  };
+};
+
+/**
+ * CallStatus fetches and displays the status of a writeContracts call.
+ *
+ * @param {Object} props - The props object.
```
Comment (Zizzamia) [inline template/web/app/new-paymaster-bundler/_components/CallStatus.tsx:14]: To help reduce all the extra comment, you can just add more explicit typing and than comment the types.

Here an example how https://github.com/coinbase/onchainkit/blob/main/src/identity/components/Avatar.tsx

File: template/web/app/new-paymaster-bundler/constants.ts [hunk 2]
```diff
@@ -1 +1,2 @@
+export const paymasterUrl = process.env.NEXT_PUBLIC_PAYMASTER_URL;
```
Comment (Zizzamia) [inline template/web/app/new-paymaster-bundler/constants.ts:1]: If is a costant to UPPER_CASE

File: template/web/src/components/SmartWallets/Capabilities.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import { useAccount } from 'wagmi';
+import { useCapabilities } from 'wagmi/experimental';
+
+/**
+ * Capabilities component displays the capabilities of the current account.
+ *
+ * @returns {JSX.Element} The rendered component displaying the capabilities or a message if none are found.
```
Comment (Zizzamia) [inline template/web/src/components/SmartWallets/Capabilities.tsx:7]: Writing comments for those component you have to learn to hit a balance, write enought to explain, but avoid to repeat yourself.

Like feel free to remove all Params and Returns, and just make descriptive the main comment. And that will be enough.

File: template/web/src/components/SmartWallets/PaymasterClient.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { ENTRYPOINT_ADDRESS_V06 } from 'permissionless';
+import { paymasterActionsEip7677 } from 'permissionless/experimental';
+import { createClient, createPublicClient, http } from 'viem';
+import { baseSepolia } from 'viem/chains';
+
+const paymasterService = process.env.NEXT_PUBLIC_PAYMASTER_URL ?? '';
+
+export const client = createPublicClient({
```
Comment (Zizzamia) [inline template/web/src/components/SmartWallets/PaymasterClient.ts:8]: why are you exporting both of them?

File: template/web/src/utils/isLocal.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import { getCurrentEnvironment } from '../store/environment';
+
+const env = getCurrentEnvironment();
+
+export default function isLocal() {
```
Comment (Zizzamia) [inline template/web/src/utils/isLocal.ts:5]: Ok, neat.
This should have tests as well.

File: template/web/app/new-paymaster-bundler/_components/CallStatus.tsx (hunk 1 above)
Comment (cpcramer) [reply template/web/app/new-paymaster-bundler/_components/CallStatus.tsx]: **`Explicit is better than implicit.`**

File: template/web/src/components/SmartWallets/PaymasterClient.ts (hunk 4 above)
Comment (cpcramer) [reply template/web/src/components/SmartWallets/PaymasterClient.ts:4]: Updated

File: template/web/app/new-paymaster-bundler/_components/CallStatus.tsx (hunk 1 above)
Comment (cpcramer) [reply template/web/app/new-paymaster-bundler/_components/CallStatus.tsx]: Added more explicit typing

File: template/web/src/utils/test/isLocal.test.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    ...actual,
+    getCurrentEnvironment: jest.fn(),
+  };
+});
+
+const mockedGetCurrentEnvironment = getCurrentEnvironment as jest.MockedFunction<
+  typeof getCurrentEnvironment
+>;
+
+describe('isLocal', () => {
```
Comment (Zizzamia) [inline template/web/src/utils/test/isLocal.test.ts:18]: nice

File: template/web/app/paymaster-bundler/constants.ts [hunk 7]
```diff
@@ -1 +1,2 @@
+export const PAYMASTER_URL = process.env.NEXT_PUBLIC_PAYMASTER_URL;
```
Comment (Zizzamia) [inline template/web/app/paymaster-bundler/constants.ts:1]: nice

**Comment by Zizzamia** [comment]
@cpcramer the PR is ready to be merged, make sure first to get all CI check pass.

**Comment by cpcramer** [comment]
> @cpcramer the PR is ready to be merged, make sure first to get all CI check pass.

Sounds good! Realizing I have a gap in TypeScript knowledge...getting more familiar with TS syntax to help better understand these errors!

File: template/web/src/store/createWagmiConfig.ts [hunk 8]
```diff
@@ -16,7 +16,6 @@ export function createWagmiConfig(rpcUrl: string, projectId?: string) {
     chains: [baseSepolia],
     connectors: [
       coinbaseWallet({
-        appChainIds: [baseSepolia.id],
```
Comment (Zizzamia) [inline template/web/src/store/createWagmiConfig.ts:19]: Oh was this the type issue? I believe this is required to have Smart Wallet working.

Can you confirm that without this Smart Wallet it is still working?

**Comment by Zizzamia** [comment]
@cpcramer what about this https://github.com/coinbase/build-onchain-apps/pull/491/files#r1606101173 ?

File: template/web/src/store/createWagmiConfig.ts (hunk 8 above)
Comment (cpcramer) [reply template/web/src/store/createWagmiConfig.ts:19]: Yes it was a type error:

`Object literal may only specify known properties, and 'appChainIds' does not exist in type '{ headlessMode?: false | undefined; version?: "4" | "3" | undefined; appName?: string | undefined; appLogoUrl?: string | null | undefined; preference?: "all" | "smartWalletOnly" | "eoaOnly" | undefined; keysUrl?: string | undefined; } | { ...; }'.ts(2353)`

Comment (cpcramer) [reply template/web/src/store/createWagmiConfig.ts:19]: `wagmi` version was bumped and now the `createConfig` function does not have the `appChainIds` property

Comment (cpcramer) [reply template/web/src/store/createWagmiConfig.ts:19]: My hunch was that `coinbaseWallet` was grabbing the `appChainId` from [`chains` ](https://github.com/coinbase/build-onchain-apps/pull/491/files#diff-849629a05d0bdce0b88b2e70e8bea52f77cb682413156c93fd260d907b27464cR16)

Comment (cpcramer) [reply template/web/src/store/createWagmiConfig.ts:19]: `coinbaseWallet` connecter documentation does not have `appChainIds` https://wagmi.sh/react/api/connectors/coinbaseWallet

**Review by 0xAlec — APPROVED** [review]

**Pull request merged**